consul/snapshot/snapshot.go

```diff
@@ -191,3 +191,25 @@ func Restore(logger *log.Logger, in io.Reader, r *raft.Raft) error {
 
 	return nil
 }
+
+// ReadMetadata takes a snapshot from the reader and returns its SnapshotMeta
+func ReadMetadata(in io.Reader) (*raft.SnapshotMeta, error) {
```

Comment:
Let's add the metadata pointer as a return value to `Verify()`. It's easy enough to ignore that in other places.